File: src/main/java/hu/frontrider/arcana/server/commands/StructureSpawnerCommand.java
package hu.frontrider.arcana.server.commands;

import hu.frontrider.arcana.worldgen.generators.TaintWineGenerator;
import net.minecraft.command.CommandException;
import net.minecraft.command.ICommandSender;
import net.minecraft.command.WrongUsageException;
import net.minecraft.command.server.CommandSetBlock;
import net.minecraft.server.MinecraftServer;
import net.minecraft.util.math.BlockPos;

/**

 */
public class StructureSpawnerCommand extends CommandSetBlock {
    /**
     * Gets the name of the command
     */
    public String getName() {
        return "spawnArcanaStructure";
    }

    /**
     * Return the required permission level for this command.
     */
    public int getRequiredPermissionLevel() {
        return 2;
    }

    /**
     * Gets the usage string for the command.
     */
    public String getUsage(ICommandSender sender) {
        return "commands.thaumic_arcana.spawn_structure.usage";
    }

    @Override
    public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException {
        if (args.length < 4) {
            throw new WrongUsageException("commands.setblock.usage");
        }
        BlockPos blockpos = parseBlockPos(sender, args, 0, false);

        switch (args[3]) {
            case "taintWine": {
                new TaintWineGenerator().generate(sender.getEntityWorld(), sender.getEntityWorld().rand, blockpos);
                break;
            }
            default:
                throw new CommandException("invalid name");
        }
    }
}

File: src/main/java/hu/frontrider/arcana/sided/server/commands/StructureSpawnerCommand.kt
package hu.frontrider.arcana.server.commands

import hu.frontrider.arcana.content.worldgen.generators.TaintWineGenerator
import net.minecraft.command.CommandBase
import net.minecraft.command.CommandException
import net.minecraft.command.ICommandSender
import net.minecraft.command.WrongUsageException
import net.minecraft.command.server.CommandSetBlock
import net.minecraft.server.MinecraftServer

/**
 *
 */
class StructureSpawnerCommand : CommandSetBlock() {
    /**
     * Gets the name of the command
     */
    override fun getName(): String {
        return "spawnArcanaStructure"
    }

    /**
     * Return the required permission level for this command.
     */
    override fun getRequiredPermissionLevel(): Int {
        return 2
    }

    /**
     * Gets the usage string for the command.
     */
    override fun getUsage(sender: ICommandSender?): String {
        return "commands.thaumic_arcana.spawn_structure.usage"
    }

    @Throws(CommandException::class)
    override fun execute(server: MinecraftServer?, sender: ICommandSender, args: Array<String>) {
        if (args.size < 4) {
            throw WrongUsageException("commands.thaumic_arcana.spawn_structure.usage")
        }
        val blockpos = CommandBase.parseBlockPos(sender, args, 0, false)

        when (args[3]) {
            "taintWine" -> {
                TaintWineGenerator().generate(sender.entityWorld, sender.entityWorld.rand, blockpos)
            }
            else -> throw CommandException("commands.thaumic_arcana.spawn_structure.invalid_name")
        }
    }
}
